user/overriding-packages.rst

```diff
@@ -0,0 +1,282 @@
+Make sure the package in the overlay has identical specifications to the version in the underlay, or only adds new ones.
+If any specification has been changed or removed then it may not be possible to override this package.
+
+How to make it easier for your users to override
+------------------------------------------------
+
+This section has advice for package authors about how to make easier for your users to use your package and override it or other pacakges.
+
+Install headers to a unique include directory
+*********************************************
+
+Install your package's headers to a unique folder rather than a shared folder.
+Say you're the author of a package ``foo``, and it has a header meant to be included like ``#include <foo/foo.hpp>``.
+Instead of installing the header to ``<prefix>/include/foo/foo.hpp``, install it to ``<prefix>/include/foo/foo/foo.hpp``.
+
+If you have a CMake package ``foo``  with the directory structure ``include/foo/foo.hpp``, then it can install its headers to a unique directory with this:
+
+.. code-block:: CMake
+
+  install(DIRECTORY include/ DESTINATION include/${PROJECT_NAME})
+
+All exported targets in your project need to export the unique include directory too.
+
+.. code-block:: CMake
+
+    target_include_directories(some_library_in_foo INTERFACE
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+      "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
+
+Dynamically link to libraries outside your package
+**************************************************
+
+If your package ``foo`` statically links to ``libbar.lib`` from package ``bar``, then users can't override ``bar`` without also overriding yours.
+Prefer dynamic linking to ``libbar.so`` instead.
+
+Handling Python entry point specifications
+******************************************
+
+If your package loads Python entry points and it encounters two specifications with the same name, then it should use the last specification returned by `entry_points() <https://docs.python.org/3/library/importlib.metadata.html#entry-points>`_.
+It should also ignore entry points that can't be loaded.
+
+Here's how to do it:
+
+.. code-block:: Python
+
+    from importlib.metadata import entry_points
+
+    # Deduplicate entry point specifications before loading
+    deduplicated_entry_points = {}
+    # When faced with duplicates, this loop keeps the last entry point found
+    for ep in entry_points()['your_group_name']:
+        deduplicated_entry_points[ep.name] = ep
+
+    for ep in deduplicated_entry_points:
+        try:
+            inst = ep.load()
+        except ImportError:
+            # Ignore entry point specifications that can't be loaded
+            pass
+
+All Known issues
+----------------
+
+Include Directory Search Order Problem
+**************************************
+
+When overriding a package, it's possible for a package to find its headers from the underlay instead of the overlay.
```

Comment:
This reads a little ambiguously to me.

```suggestion
When overriding a package, it's possible for another package to find that package's headers from the underlay instead of the overlay.
```